core/node/reorg_detector/src/lib.rs

```diff
@@ -165,6 +173,21 @@ impl MainNodeClient for Box<DynClient<L2>> {
         };
         Ok(batch.base.root_hash.ok_or(MissingData::RootHash))
     }
+
+    async fn l1_batch_commitment(
```

Comment:
I'm somewhat hesitant about this addition of 1 RPC call per batch. It might make debugging easier, but this mismatch would also be found during commit transaction finality status processing. (there would be mismatch between event logs and local state)

Reply:
that's right, but it won't be in reorg detector....we just can't find the diverged block